dash/_patch.py

```diff
@@ -0,0 +1,140 @@
+
+    def __setattr__(self, key, value):
+        if key in ("_location", "_operations"):
+            self.__dict__[key] = value
+        else:
+            self.__setitem__(key, value)
+
+    def __delattr__(self, item):
+        self.__delitem__(item)
+
+    def __setitem__(self, key, value):
+        validate_slice(key)
+        if value is _noop:
+            # The += set themselves.
+            return
+        self._operations.append(
+            _operation(
+                "Assign",
+                self._location + [key],
+                value=value,
+            )
+        )
+
+    def __delitem__(self, key):
+        validate_slice(key)
+        self._operations.append(_operation("Delete", self._location + [key]))
+
+    def __iadd__(self, other):
+        if isinstance(other, (list, tuple)):
+            self.extend(other)
+        else:
+            self._operations.append(_operation("Add", self._location, value=other))
+        return _noop
+
+    def __isub__(self, other):
+        self._operations.append(_operation("Sub", self._location, value=other))
+        return _noop
+
+    def __imul__(self, other):
+        self._operations.append(_operation("Mul", self._location, value=other))
+        return _noop
+
+    def __itruediv__(self, other):
+        self._operations.append(_operation("Div", self._location, value=other))
+        return _noop
+
+    def __ior__(self, other):
+        self.update(E=other)
+        return _noop
+
+    def append(self, item):
+        """Add the item to the end of a list"""
+        self._operations.append(_operation("Append", self._location, value=item))
+
+    def prepend(self, item):
+        """Add the item to the start of a list"""
+        self._operations.append(_operation("Prepend", self._location, value=item))
+
+    def insert(self, index, item):
+        """Add the item at the index of a list"""
+        self._operations.append(
+            _operation("Insert", self._location, value=item, index=index)
+        )
+
+    def clear(self):
+        """Remove all items in a list"""
+        self._operations.append(_operation("Clear", self._location))
+
+    def reverse(self):
+        """Reversal of the order of items in a list"""
+        self._operations.append(_operation("Reverse", self._location))
+
+    def extend(self, item):
+        """Add all the items to the end of a list"""
+        if not isinstance(item, (list, tuple)):
+            raise TypeError(f"{item} should be a list or tuple")
+        self._operations.append(_operation("Extend", self._location, value=item))
+
+    def remove(self, item):
+        """filter the item out of a list on the frontend"""
+        self._operations.append(_operation("Remove", self._location, value=item))
+
+    def update(self, E=None, **F):
+        """Merge a dict or keyword arguments with another dictionary"""
+        if E is not None:
+            value = E
+        else:
+            value = dict(F)
```

Comment:
kind of odd usage, but it turns out you can use both of these together.
```py
>>> a={}
>>> a.update({1:2}, a=4,b=5)
>>> a
{1: 2, 'a': 4, 'b': 5}
```
So maybe:
```suggestion
        value = E or {}
        value.update(F)
```